### DISCUSSION TITLE
MuPDF version used when different PyMuPDF versions installed in different Python environments?

### DISCUSSION THREAD

**Discussion by ishineko** [pymupdf/PyMuPDF#1437] [Q&A] [answered]
This is just a question/clarification:

If we have different versions of PyMuPDF installed in separate Python environments, what is the version of MuPDF that they each use? Do they share a common install of MuPDF, or is each PyMuPDF installed on each separate environment running their own version of MuPDF?

We ask because we want to keep archived versions of Python environments for audit purposes, but since PyMuPDF is using MuPDF behind the scenes, we're wondering what happens when a new version is installed on a different environment.

Thanks for the great tool, JorjMcKie!

**Comment by JorjMcKie** [comment, marked as answer]
PyMuPDF is not dependent on any MuPDF installation that may otherwise exist: the required MuPDF code is bound into its own binary.
Hence multiple PyMuPDF versions may also exist in their resp. environments.
